### ISSUE TITLE
Tk problems on macOS Catalina

### ISSUE THREAD

**Issue by simonjwright** [simonjwright/tcladashell#4]
The Xcode 11.6 SDK has two changes that impact us: first, there are no X11 includes (they used to be in the Tk framework), and second, Tk is now deprecated.

In both cases, one solution might be to use ActiveTcl from [ActiveState](https://platform.activestate.com/).

Failing that, you can download and install [XQuartz](https://www.xquartz.org) to solve the first problem (adjusting the `X11_INCLUDE` setting in `setup.tcl` to `-I/opt/X11/include`).

The second problem surfaced when I tried to use the FSF GCC 10.1.0 compiler from [Sourceforge](https://sourceforge.net/projects/gnuada/files/GNAT_GCC%20Mac%20OS%20X/10.1.0/native/); it didn't understand the deprecation macros,
```
$ make
tclsh tcl_record_sizes.tcl >tcl_record_sizes.c
gcc -I/usr/include -o tcl_record_sizes tcl_record_sizes.c
./tcl_record_sizes >tcl_record_sizes.ads
gprbuild -p -P ../tash.gpr -XLIBRARY_TYPE=static
Compile
   [c]            tkmacro.c
In file included from /Users/simon/tcladashell/src/tkmacro.c:36:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/tk.h:27:22: error: expected declaration specifiers or '...' before string constant
   27 | API_DEPRECATED_BEGIN("Tk API deprecated.", macos(10.0, 10.15))
      |                      ^~~~~~~~~~~~~~~~~~~~
```

**Comment by simonjwright** [comment]
It turned out that the deprecation macro problem was fixed by re-installing the compiler! Installing GCC involves 'fixing the includes' - rewriting those of the system include files that are non-standard (by GCC’s lights, anyway). I had installed GCC 10.1.0 with Xcode 11.5 (I think); since then, Xcode 11.6 had been installed, and GCC’s changes were now both unnecessary and wrong.

**Comment by thindil** [comment]
Some suggestion about Tcl/Tk deprecation: How about using the Homebrew version? It should simplify maintaining Tcl/Tk and its dependencies. Homebrew provides two versions:
* https://formulae.brew.sh/formula/tcl-tk Vanilla Tcl/Tk with standard permissive license, compile from sources
* https://formulae.brew.sh/cask/tcl#default ActiveState version for personal use only (my guess)

**Comment by simonjwright** [comment]
I suppose it might, but I don't run Homebrew (or Macports) so would have trouble supporting either.

I may be being partly driven by recollections of incompatibilities with Cygwin's tcl/tk vs ActiveState's (Cygwin's failed completely).

**Comment by thindil** [comment]
Ah, I see. Then yes, probably ActiveState can be the best option. Just there will be a problem with keeping it up to date (it don't have any autoupdater if I remember correctly). License should be not a problem as long as will not need to redistribute Tcl/Tk.

Or, maybe something from here: http://tcl.tk/software/tcltk/bindist.html (binary distributions of Tcl/Tk). For MacOS only this "single file executable" options can work.

**Comment by simonjwright** [comment]
Fixed in Release 20200802 (d73de1e).
